src/tools/cherry-pick.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env zsh
+set -euo pipefail
+IFS=$'\n\t'
+
+print_error() {
+  echo "Error: \`$1\` is not a valid commit. To debug, run:"
+  echo
+  echo "    git rev-parse --verify $1"
+  echo
+}
+
+full_sha() {
+  git rev-parse \
+    --verify \
+    --quiet \
+    "$1^{object}" || print_error $1
+}
+
+commit_message_with_backport_note() {
+  message=$(git log --format=%B -n 1 $1)
+  echo $message | awk "NR==1{print; print \"\n(backport-of: $1)\"} NR!=1"
+}
+
+sha=$(full_sha $1)
+git cherry-pick $sha
+git commit \
+  --amend \
+  --file <(commit_message_with_backport_note $sha)
```

Comment:
Could you refactor this to take multiple parameters (and just cherry-pick each in turn, just like the normal cherry pick)?